larpandoracontent/LArHelpers/LArHitWidthHelper.cc

```diff
@@ -0,0 +1,411 @@
+{
+    CartesianPointVector constituentHitPositionVector;
+
+    for(const ConstituentHit &constituentHit : constituentHitVector)
+    {
+        constituentHitPositionVector.push_back(constituentHit.GetPositionVector());
+    }
+
+    return constituentHitPositionVector;
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+bool LArHitWidthHelper::SortByHigherXExtrema(const Cluster *const pLhs, const Cluster *const pRhs)
+{
+
+    // Get the map of cluster to constitutent hit parameters
+    LArHitWidthHelper::ClusterToParametersMapStore* pClusterToParametersMapStore = LArHitWidthHelper::ClusterToParametersMapStore::Instance();
+    LArHitWidthHelper::ClusterToParametersMap* pClusterToParametersMap = pClusterToParametersMapStore->GetMap();
+
+    if(pClusterToParametersMap->empty())
+        throw StatusCodeException(STATUS_CODE_NOT_INITIALIZED);
+
+    LArHitWidthHelper::ClusterToParametersMap::const_iterator lhsIter(pClusterToParametersMap->find(pLhs));
+    LArHitWidthHelper::ClusterToParametersMap::const_iterator rhsIter(pClusterToParametersMap->find(pRhs));
+
+    if(lhsIter == pClusterToParametersMap->end() || rhsIter == pClusterToParametersMap->end())
+        throw StatusCodeException(STATUS_CODE_NOT_FOUND);
+
+
+    CartesianVector lhsHigherXExtrema(lhsIter->second.GetHigherXExtrema()), rhsHigherXExtrema(rhsIter->second.GetHigherXExtrema());
+
+    return (lhsHigherXExtrema.GetX() < rhsHigherXExtrema.GetX());
+
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+void LArHitWidthHelper::GetExtremalCoordinatesX(const ConstituentHitVector &constituentHitVector, CartesianVector &lowerXCoordinate, CartesianVector &higherXCoordinate)
+{
+
+    CartesianPointVector constituentHitPositionVector(GetConstituentHitPositionVector(constituentHitVector));
+    GetExtremalCoordinatesX(constituentHitPositionVector, lowerXCoordinate, higherXCoordinate);
+
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+
+void LArHitWidthHelper::GetExtremalCoordinatesX(const CartesianPointVector &constituentHitPositionVector, CartesianVector &lowerXCoordinate, CartesianVector &higherXCoordinate)
+{
+
+    // Find the extremal values of the X, Y and Z coordinates
+    float xMin(+std::numeric_limits<float>::max());
+    float yMin(+std::numeric_limits<float>::max());
+    float zMin(+std::numeric_limits<float>::max());
+    float xMax(-std::numeric_limits<float>::max());
+    float yMax(-std::numeric_limits<float>::max());
+    float zMax(-std::numeric_limits<float>::max());
+
+    for (CartesianPointVector::const_iterator pIter = constituentHitPositionVector.begin(), pIterEnd = constituentHitPositionVector.end(); pIter != pIterEnd; ++pIter)
```

Comment:
this function is one from the LArClusterHelper which i copied but since have removed the copy.. would you like me to make these changes to the original in LArClusterHelper?

Reply:
No, I think for now you don't need to worry about making stylistic changes to existing parts of Pandora.